safe_transaction_service/history/views.py

```diff
@@ -822,6 +859,88 @@ def delete(self, request, delegate_address, *args, **kwargs):
             return Response(status=status.HTTP_404_NOT_FOUND)
 
 
+class TransferView(GenericAPIView):
+    serializer_class = serializers.TransferWithTokenInfoResponseSerializer
+    pagination_class = None
+
+    def get_erc20_erc721_transfer(self, transaction_hash: str, log_index: str):
+        """
+        Search ERCTransfer by transaction_hash and log_index event.
+
+        :param transaction_hash: ethereum transaction hash
+        :param log_index: event log index
+        :return: ERC20Transfer object
+        """
+        erc20_queryset = self.filter_queryset(
+            ERC20Transfer.objects.filter(
+                ethereum_tx=transaction_hash, log_index=log_index
+            ).token_txs()
+        )
+        erc721_queryset = self.filter_queryset(
+            ERC721Transfer.objects.filter(
+                ethereum_tx=transaction_hash, log_index=log_index
+            ).token_txs()
+        )
+        return ERC20Transfer.objects.token_transfer_values(
+            erc20_queryset, erc721_queryset
+        )
+
+    def get_ethereum_transfer(self, transaction_hash: str, trace_address: str):
+        """
+        Search an ethereum transfer by transaction hash and trace address
+
+        :param transaction_hash: ethereum transaction hash
+        :param trace_address: ethereum trace address
+        :return: InternalTx object
+        """
+        ether_queryset = self.filter_queryset(
+            InternalTx.objects.ether_txs().filter(
+                ethereum_tx=transaction_hash, trace_address=trace_address
+            )
+        )
+        return InternalTx.objects.ether_txs_values(ether_queryset)
+
+    def get_queryset(self, transfer_id):
```

Comment:
We must check that `transfer_id` follows the spec, either here or by a regular expression on the route.

Also, a test should be added for this.

At last, missing type for `transfer_id` and return type